src/app/clusters/ota-requestor/BDXDownloader.cpp

```diff
@@ -118,6 +118,9 @@ CHIP_ERROR BDXDownloader::BeginPrepareDownload()
     VerifyOrReturnError(mImageProcessor != nullptr, CHIP_ERROR_INCORRECT_STATE);
 
     mPrevBlockCounter = 0;
+
+    // Note that due to the nature of this timer function, the actual time taken to detect a stalled BDX connection might be
+    // anywhere in the range of [mTimeout, 2*mTimeout)
```

Comment:
I'm not sure I understand why it could take up to twice the amount of timeout?

Reply:
The timer handler only gets called once every mTimeout seconds to check if the connection is still active...
```

T0 -> Timer starts
T0+1 -> 1st block is received
T0+2 -> BDX connection stalled due to provider disconnecting
...
T0+mTimeout -> Timer kicks and TransferTimeout handler is called which updates the curBlockCounter and restarts timer
...
T0+(2*mTimeout) -> Timer kicks and TransferTimeout handler sees curBlockCounter hasn't changed and kills the connection.
```

Reply:
If `curBlockCounter` updated, then that is not the point of connection stall? Anyway, not important pertaining to this fix.

Reply:
Sorry that was a typo- I meant "mPrevBlockCounter" is updated at T0+mTimeout ('cos one block got transferred).
So, even though the connection stalled at T0+2, we do not discover it until T0+(2*mTimeout).